### PR TITLE
Implement a verifier for the client_02

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: core/src/ibc/client_02/manager.rs [hunk 1]
```diff
@@ -72,12 +76,54 @@ impl<'a> Manager<'a> {
...
+        connection_identifier: IdentifierSlice,
+        connection_end: &ConnectionEnd,
+    ) -> Result<(), String> {
+        let path = format!("connections/{}", connection_identifier);
+        let client_state = self.query(&id)?;
+        if client_state.raw.number < proof_height {
+            return Err("Invalid proof height".to_owned())
+        }
+        let consensus_state = self.query_consensus_state(&id, proof_height)?;
+        let proof_dec: CommitmentProofCounter = rlp::decode(&proof).map_err(|_| "Illformed proof")?;
```
Comment (junha1) [inline core/src/ibc/client_02/manager.rs]: This error message will be advanced to include the name of verify function after I add all of `verify_*()` functions and make some reusable routine out of them.

File: core/src/ibc/client_02/manager.rs [hunk 2]
```diff
@@ -72,12 +76,54 @@ impl<'a> Manager<'a> {
...
+            &consensus_state.state_root,
+            &proof_dec,
+            CommitmentPathCounter {
+                raw: path,
+            },
+            value_dec,
+        ) {
+            Ok(())
+        } else {
+            Err("Invalid proof".to_owned())
```
Comment (junha1) [inline core/src/ibc/client_02/manager.rs]: Any better idiomatic way of doing this? @majecty

Comment (majecty) [reply core/src/ibc/client_02/manager.rs]: I don't know. If you find a better way to do it, please share it with me.

**Comment by junha1** [comment]
@majecty I think it's ready to get review.

File: core/src/ibc/client_02/mod.rs [hunk 3]
```diff
@@ -15,11 +15,14 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 pub mod client;
+mod commitment_23;
 mod manager;
 pub mod types;
+
```
Comment (majecty) [inline core/src/ibc/client_02/mod.rs]: Please remove the unrelated code change.
Please check your PR once before you request review.

Comment (junha1) [reply core/src/ibc/client_02/mod.rs]: I'll fix the new line in line 25, but I think here is a valid change I think. Don't we separate `mod` and `use`? @majecty

Comment (majecty) [reply core/src/ibc/client_02/mod.rs]: Please fix it in a new commit.

Comment (junha1) [reply core/src/ibc/client_02/mod.rs]: You mean a commit that just adds a single new line?

Comment (majecty) [reply core/src/ibc/client_02/mod.rs]: Yes. That is what I mean.

Comment (junha1) [reply core/src/ibc/client_02/mod.rs]: @majecty fixed.

Comment (majecty) [reply core/src/ibc/client_02/mod.rs]: Please use more descriptive commit message.
Like this: "Add a new line between mod statement and use statement.

Comment (junha1) [reply core/src/ibc/client_02/mod.rs]: @majecty done!

**Comment by majecty** [comment]
Let's merge this PR after the clippy and rust-fmt tests.
